pkg/tests/run.go

```diff
@@ -16,83 +17,39 @@ func Run(
 	metrics map[string]*vegeta.Metrics,
 	outputDirectory string,
 	duration time.Duration,
-	connection *sdk.Connection) error {
+	connection *sdk.Connection,
+	viper *viper.Viper) error {
 
 	// testId provides a common value to associate all output data from running
 	// the full test suite with a single test run.
 	testID := uuid.NewV4().String()
 
-	// Specify Test Cases
-	// They are written this way to re-use functionality where possible and
-	// hopefully make it easier to modify and/or extend given the declarative
-	// style.
-	tests := []helpers.TestOptions{
-		{
-			TestName: "self-access-token",
-			Path:     helpers.SelfAccessTokenEndpoint,
-			Method:   http.MethodPost,
-			Rate:     helpers.SelfAccessTokenRate,
-			Handler:  handlers.TestStaticEndpoint,
-		},
-		{
-			TestName: "list-subscriptions",
-			Path:     helpers.ListSubscriptionsEndpoint,
-			Method:   http.MethodGet,
-			Rate:     helpers.ListSubscriptionsRate,
-			Handler:  handlers.TestStaticEndpoint,
-		},
-		{
-			TestName: "access-review",
-			Path:     helpers.AccessReviewEndpoint,
-			Method:   http.MethodPost,
-			Body:     "{\"account_username\": \"rhn-support-tiwillia\", \"action\": \"get\", \"resource_type\": \"Subscription\"}",
-			Rate:     helpers.AccessReviewRate,
-			Handler:  handlers.TestStaticEndpoint,
-		},
-		{
-			TestName: "register-new-cluster",
-			Path:     helpers.ClusterRegistrationEndpoint,
-			Method:   http.MethodPost,
-			Rate:     helpers.RegisterNewClusterRate,
-			Handler:  handlers.TestRegisterNewCluster,
-		},
-		{
-			TestName: "create-cluster",
-			Rate:     helpers.CreateClusterRate,
-			Handler:  handlers.TestCreateCluster,
-		},
-		{
-			TestName: "list-clusters",
-			Rate:     helpers.ListClustersRate,
-			Handler:  handlers.TestListClusters,
-		},
-		{
-			TestName: "get-current-account",
-			Path:     helpers.GetCurrentAccountEndpoint,
-			Method:   http.MethodGet,
-			Rate:     helpers.GetCurrentAccountRate,
-			Handler:  handlers.TestStaticEndpoint,
-		},
-	}
-
-	for i := range tests {
-
+	for _, t := range tests {
+		if viper.InConfig(t.TestName) {
+			f := viper.GetInt(fmt.Sprintf("%s.freq", t.TestName))
+			p := viper.GetString(fmt.Sprintf("%s.per", t.TestName))
+			per, err := time.ParseDuration("1" + p)
```

Comment:
Would you mind adding a comment above this block of code describing what you're doing here: 
> Generating the per-test duration and frequency using the information specified in the config file or defaulting to `helpers.DefaultRate`."

The reason is that this code is a tiny complex and would not be clear in a couple of months from now to someone new coming into this code base.

Reply:
sure